### PR TITLE
Consume arcade packages in CoreFx

### PR DESCRIPTION
**Pull request by wtgodbe**
Consumes `Microsoft.DotNet.GenFacades`, `Microsoft.DotNet.GenApi`, and `Microsoft.DotNet.ApiCompat` from Arcade.

CC @weshaggard

### CODE REVIEW

**Comment by wtgodbe** [comment]
Current state: trying to get just `GenFacades` working. I'm getting errors that look like the following:

> System.Threading.Tasks.Extensions.cs(154,120): error CS0436: The type 'ValueTaskSourceOnCompletedFlags' in 'E:\code\corefx\src\System.Threading.Tasks.Extensions\ref\System.Threading.Tasks.Extensions.cs' conflicts with the imported type 'ValueTaskSourceOnCompletedFlags' in 'System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in 'E:\code\corefx\src\System.Threading.Tasks.Extensions\ref\System.Threading.Tasks.Extensions.cs'.

**Comment by wtgodbe** [comment]
Looks like https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets#L17-L20 isn't getting hit, because `IsReferenceAssembly` is set by this file in Buildtools (which gets imported by `Build.Common.Props`)

@weshaggard I'm thinking the best fix is to set this property in https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets, do you agree?

**Comment by weshaggard** [comment]
>@weshaggard I'm thinking the best fix is to set this property in https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets, do you agree?

I don't think we should be setting IsReferenceAssembly inside of the GenFacades targets file. I assume this will work if you rearrange the order and so IsReferenceAssembly will be correctly set before you import the GenFacades?

As for IsCompatAssembly I don't even believe that is used any longer and so I wouldn't worry about that.

**Comment by wtgodbe** [comment]
Correct, it will work if I rearrange the order, but it wouldn't work for somebody who wasn't depending on the property being set by Buildtools (i.e. another user might run into this problem if using the package in isolation)

**Comment by weshaggard** [comment]
>Correct, it will work if I rearrange the order, but it wouldn't work for somebody who wasn't depending on the property being set by Buildtools (i.e. another user might run into this problem if using the package in isolation). As it is right now, IsReferenceAssembly has no meaning in the GenFacades targets file, and will only work if some external thing happens to set it correctly

For those folks they will have to disable the warning in their project. I'd be fine if you do the warning disable condition in the Directory.Builds.props in corefx as well and remove it from genfacades targets.

**Comment by wtgodbe** [comment]
There are a few other `PropertyGroup`s in GenFacades.targets that depend on `IsReferenceAssembly` (e.g. https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets#L92), it seems wrong to me to have those depend on `Build.Common.props`, since we eventually want to move away from that. Are you suggesting we depend on `Build.Common.props` for now, with the plan of later having some central place in Arcade for such common properties?

**Comment by weshaggard** [comment]
>There are a few other PropertyGroups in GenFacades.targets that depend on IsReferenceAssembly (e.g. https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets#L92), it seems wrong to me to have those depend on Build.Common.props, since we eventually want to move away from that. Are you suggesting we depend on Build.Common.props for now, with the plan of later having some central place in Arcade for such common properties?

No I don't want the GenFacades target to directly import Build.Common.props but if it where to just be able disabling a warning that could go in Build.Common.props instead of the GenFacades targets. However given there are other dependencies on IsReferenceAssembly I think you just need to switch the order around.

**Comment by wtgodbe** [comment]
In its current state, the build is failing in APICompat. Will try using Arcade's APICompat to see if ordering fixes the issue.

**Comment by wtgodbe** [comment]
APICompat now fails with:

>   Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

**Comment by wtgodbe** [comment]
Looking at the binlogs, I see that with these changes, we run `Build`, then `Compile`, then `CoreBuild` (sequentially) as the default targets. `Build` just invokes `BeforeBuild`, and nothing else. The errors happen in the `CSC` task, which is called by `CoreCompile`, which is called by `Compile`.

In master, however, the only default Target is `Build`, which invokes many sub-targets (`BeforeBuild`, `CoreBuild`, and `SignFiles`). Inside of `CoreBuild`, several other targets are called, including `Compile`, which eventually calls `CoreCompile`, which eventually invokes `CSC`. 

The actual calls to `CSC` appear identical across both instances (other than the errors being produced), but I wonder if the problem lies in the different path each takes to reach CSC. Right now I'm not sure why we'd be following a different path of Targets, which is what I'm trying to figure out.

**Comment by wtgodbe** [comment]
For some reason in my branch, `https://github.com/dotnet/arcade/blob/e9ad66fd36b9b4f11b8125064134ded0e76bc307/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets#L45` is not being honored. `TargetsTriggeredByCompilation` is updated to include `FillPartialFacadeUsingTask`, but that target never gets called, so `System.Threading.Tasks.Extensions` winds up not getting the type-forwards it should have. Looking at the binlogs, it appears that we don't actually call `CSC` in the build of `System.Threading.Tasks.Extensions` in my branch, whereas in master, we do call `CSC` which triggers  `FillPartialFacadeUsingTask` & the other `TargetsTriggeredByCompilation`

**Comment by weshaggard** [comment]
>For some reason in my branch, https://github.com/dotnet/arcade/blob/e9ad66fd36b9b4f11b8125064134ded0e76bc307/src/Microsoft.DotNet.GenFacades/build/Microsoft.DotNet.GenFacades.targets#L45 is not being honored. TargetsTriggeredByCompilation is updated to include FillPartialFacadeUsingTask, but that target never gets called, so System.Threading.Tasks.Extensions winds up not getting the type-forwards it should have. Looking at the binlogs, it appears that we don't actually call CSC in the build of System.Threading.Tasks.Extensions in my branch, whereas in master, we do call CSC which triggers FillPartialFacadeUsingTask & the other TargetsTriggeredByCompilation

Try and see if you can figure out which CSharp.targets is getting imported in both cases. Try doing a pp log for both to ensure they are the same type of project.

**Comment by wtgodbe** [comment]
My branch & master are both importing `corefx\Tools\dotnetcli\sdk\2.1.401\Microsoft.CSharp.targets`

**Comment by wtgodbe** [comment]
Looks like I may have just needed to do a clean build - my new log shows that `FillPartialFacadeUsingTask` is being executed, and the C# errors are gone. I now have:

> E:\code\corefx\Tools\sign.targets(113,5): error : E:\code\corefx/bin/obj/AnyOS.AnyCPU.Debug/System.Security.Cryptography.OpenSsl/netcoreapp/System.Security.Cryptography.OpenSsl.dll: I/O error reading or writing PE file: Could not find file 'E:\code\corefx\bin\obj\AnyOS.AnyCPU.Debug\System.Security.Cryptography.OpenSsl\netcoreapp\System.Security.Cryptography.OpenSsl.dll'. [E:\code\corefx\src\System.Security.Cryptography.OpenSsl\src\System.Security.Cryptography.OpenSsl.csproj]

**Comment by ViktorHofer** [comment]
I noticed the OpenSsl ones also in master as a warning when building certain configurations (I think it was uap). Maybe related.

**Comment by wtgodbe** [comment]
In `illink.targets`, in the `_CheckForILLinkTrimAssemblyOutputs` target, we define ItemGroups for pretrimmed assemblies, and assemblies that were trimmed & outputted by the `ILLinkTrimAssembly` target:

https://github.com/dotnet/corefx/blob/master/illink.targets#L148-L161

What's supposed to happen (I believe) is that we output a .dll to `bin\obj\AnyOS.AnyCPU.Debug\{xyz}\netcoreapp\{xyz}.dll` - then the `.dll` gets moved to a folder named `PreTrim` relative to its old location, by https://github.com/dotnet/corefx/blob/master/illink.targets#L91-L92. Then `ILLinkTrimAssembly` does _something_, and copies the .dll back to its original location. In master, both 'PreTrimmedItem` and `TrimmedItem` get populated, while in my branch, `TrimmedItem` is left empty (as the `.dll` was not copied back to its original location). 

So it appears something in the `ILLink` task is going wrong, at:

https://github.com/dotnet/corefx/blob/master/illink.targets#L109-L114

**Comment by wtgodbe** [comment]
The calls to ILLink are identical in my branch and in master, but the `ystem.Security.Cryptography.OpenSsl.dll's` are not - the one in my branch has several API entries that the one in master doesn't.

File: Directory.Build.targets [hunk 1]
```diff
@@ -125,6 +125,11 @@
 
   <Import Project="$(ToolsDir)/Build.Common.targets" Condition="Exists('$(ToolsDir)/Build.Common.targets')" />
 
+  <!-- Import Arcade targets -->
+<!--  <Import Project="$(PackagesDir)/$(GenAPIPackage.ToLower())/$(GenAPIPackageVersion)/build/$(GenAPIPackage).targets" /> -->
```
Comment (weshaggard) [inline Directory.Build.targets]: You need to add the notsupportes support to fix the openssl issue.

Comment (wtgodbe) [reply Directory.Build.targets]: @weshaggard you think it's an ordering issue? In the current state we're importing notsupported.targets inside of build.common.targets, while the other 2 are coming from here

**Comment by wtgodbe** [comment]
When opting into Arcade's not supported targets rather than buildtools', I'm getting errors like the following (before hitting the point where the previous errors would occur):

> EXEC : error : Failed to load any assemblies from 'E:\code\corefx\bin\AnyOS.AnyCPU.Debug\binplacePackages\netstandard\binplacePackages.dll' [E:\code\corefx\external\binplacePackages\binplacePackages.depproj]

I get the same for the `ILLink` and `docs` depproj's. It looks like the depproj's are trying to call GenApi on files that shouldn't exist:

> E:\code\corefx\packages\microsoft.dotnet.genapi\1.0.0-beta.18475.1\build\Microsoft.DotNet.GenApi.targets(42,5): error MSB3073: The command ""E:\code\corefx\Tools/dotnetcli/dotnet" "E:\code\corefx\packages\microsoft.dotnet.genapi\1.0.0-beta.18475.1\build\\..\tools\netcoreapp2.1\Microsoft.DotNet.GenAPI.dll" -assembly:"E:\code\corefx\bin\AnyOS.AnyCPU.Debug\binplacePackages\netstandard\binplacePackages.dll" -libPath:"E:\code\corefx\packages\System.Memory\4.5.1\ref\netstandard2.0" -out:"E:\code\corefx\bin\AnyOS.AnyCPU.Debug\binplacePackages\netstandard\binplacePackages.cs" " exited with code 1. [E:\code\corefx\external\binplacePackages\binplacePackages.depproj]

There is no `bin\AnyOS.AnyCPU.Debug\binplacePackages\netstandard\binplacePackages.dll` built by master

**Comment by weshaggard** [comment]
You need to condition https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets#L19 so we don't run GenerateReferenceSource for all the projects.

**Comment by wtgodbe** [comment]
Looks like we shouldn't be calling `GenerateReferenceAssemblySource` (the GenAPI target) on the .depproj's. We don't call it in master, while in this branch, it gets called as part of `PrepareForRun` due to https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets#L19-L22

**Comment by wtgodbe** [comment]
> You need to condition https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets#L19 so we don't run GenerateReferenceSource for all the projects.

How do I know which projects should run the target? Looks like previously we only ran the target in https://github.com/dotnet/buildtools/blob/6736870b84e06b75e7df32bb84d442db1b2afa10/src/GenAPI.Desktop/GenAPI.Desktop.csproj, but I'm not sure how to check who was & wasn't using the Desktop project in buildtools

**Comment by weshaggard** [comment]
>How do I know which projects should run the target? Looks like previously we only ran the target in https://github.com/dotnet/buildtools/blob/6736870b84e06b75e7df32bb84d442db1b2afa10/src/GenAPI.Desktop/GenAPI.Desktop.csproj, but I'm not sure how to check who was & wasn't using the Desktop project in buildtools. RunningOnCore != true?

For now just add a condition and it isn't set to true don't include it and then we can figure out what conditions it is interesting for. I'm not sure anyone is currently depending on that target running automatically.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/arcade/pull/859

**Comment by wtgodbe** [comment]
This now builds successfully with the changes from https://github.com/dotnet/arcade/pull/859! We still shouldn't merge without waiting for a new Arcade package though. @weshaggard any other sanity checks I should do in the meantime?

File: dependencies.props [hunk 2]
```diff
@@ -73,6 +73,17 @@
     <PublishSymbolsPackageVersion>1.0.0-beta-63112-02</PublishSymbolsPackageVersion>
   </PropertyGroup>
 
+  <!-- Packages to consume from Arcade -->
+  <PropertyGroup>
+    <ApiCompatPackage>Microsoft.DotNet.ApiCompat</ApiCompatPackage>
+    <GenAPIPackage>Microsoft.DotNet.GenApi</GenAPIPackage>
+    <GenFacadesPackage>Microsoft.DotNet.GenFacades</GenFacadesPackage>
+
+    <ApiCompatPackageVersion>1.0.0-beta.18475.1</ApiCompatPackageVersion>
```
Comment (weshaggard) [inline dependencies.props]: These should go in eng\version.props

**Comment by wtgodbe** [comment]
Updated to use eng\versions.props. @weshaggard does the formatting look correct?

File: eng/Versions.props [hunk 3]
```diff
@@ -4,4 +4,16 @@
     <VersionPrefix>1.0.0</VersionPrefix>
     <PreReleaseVersionLabel>prerelease</PreReleaseVersionLabel>
   </PropertyGroup>
+  <!-- Package names -->
+  <PropertyGroup>
+    <ApiCompatPackage>Microsoft.DotNet.ApiCompat</ApiCompatPackage>
```
Comment (wtgodbe) [inline eng/Versions.props]: Will update these names to match the format of the `PackageVersion` names

**Comment by wtgodbe** [comment]
Linux failures are probably a casing issue. Looking

**Comment by wtgodbe** [comment]
> microsoft.dotnet.genfacades\1.0.0-beta.18475.8\build\Microsoft.DotNet.GenFacades.targets(97,5): error : error: Product Version could not be parsed: 1.0.24212.01. Commit Hash: 9688ddbb62c04189cac4c4a06e31e93377dccd41 [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Reflection.TypeExtensions\src\System.Reflection.TypeExtensions.csproj]

This is https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/VersionResourceSerializer.cs#L93

**Comment by wtgodbe** [comment]
This is happening because of https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Version.cs#L340. Essentially, our logic says that the first three periods mark the end of the major, minor, and build numbers. If there are any other periods in the string, we throw an exception. We only pass everything before the first whitespace character to `Version.TryParse()` (see https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.GenFacades/VersionResourceSerializer.cs#L92), but since the product version in question has `1.0.24212.01.` before the first whitespace character, that fourth period causes the exception to be thrown. I guess the choices here are to strip any trailing periods from what we pass to `TryParse`, or to modify the Version parsing logic in CoreFx to not care about any periods after the first three. I prefer the second option, since we could potentially have any weird `ProductVersion` string in a .dll. @jkotas do you know why we must throw an exception when there are more than 3 periods in the Version string? See my comment above for an example of a Version string produced by CoreFx that doesn't conform to this standard, but that does represent what should be a valid, parseable version

**Comment by jkotas** [comment]
> See my comment above for an example of a Version string produced by CoreFx that doesn't conform to this standard, but that does represent what should be a valid, parseable version

I do not see what makes you think that it is a valid parsable .NET version. .NET was never able to parse version strings with extra dot at the end.

**Comment by wtgodbe** [comment]
> I do not see what makes you think that it is a valid parsable .NET version. .NET was never able to parse version strings with extra dot at the end.

Yes, my question is, why can't we parse a version with an extra period at the end? Is there a reason we choose to throw an exception in that case?

**Comment by wtgodbe** [comment]
Also, as far as I can tell, buildtools shouldn't be producing a PackageVersion with format `x.y.z.w. commit hash: xxxxxx`. Still trying to figure out where that could be coming from

https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L313-L321

**Comment by wtgodbe** [comment]
> The parsers do not tolerate extra junk. It is general design pattern. If we were to start tolerating extra junk, where would we stop? E.g. should a version string with an extra .0 at the end be valid too?

That makes sense, but if that's the case, why do we append extra stuff to the end of the Product Version String here? https://github.com/dotnet/buildtools/blob/4642e67685b9c21c4d5160ea7665843a48ed8155/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L104

In any case, this isn't the place that's creating my Product Version string, so I think the solution for my problem is to find who's creating the bad String & to modify the logic there

**Comment by wtgodbe** [comment]
Looks like the weird String is coming from here: https://github.com/dotnet/buildtools/blob/release/1.1.0/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L150

I'm going to try to change our logic so that we use the ProductVersion rather than the ProductVersionString.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/arcade/blob/04f4b34837651ad724f48362285f0a5d2ffa4a90/src/Microsoft.DotNet.GenFacades/GenFacades.cs#L668 doesn't set a suitable value for `ProductVersion`. It looks like we're getting the version info by calling https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Windows.cs#L27, which calls https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/Common/src/Interop/Windows/version/Interop.GetFileVersionInfoEx.cs#L12-L13, so I'm not sure what we can do here to return the ProductVersion rather than the ProductVersionString.

**Comment by weshaggard** [comment]
>so I'm not sure what we can do here to return the ProductVersion rather than the ProductVersionString. @weshaggard do you think there's another way around this?

How about we try to parse the ProductVersion and if that fails we fallback to using the FileVersion instead.

**Comment by wtgodbe** [comment]
> How about we try to parse the ProductVersion and if that fails we fallback to using the FileVersion instead.

That should solve the current problem, since the File Version was parsed successfully in this case - but isn't there still the possibility that both are malformed? 

https://github.com/dotnet/buildtools/blob/release/1.1.0/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L150

https://github.com/dotnet/buildtools/blob/release/1.1.0/src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets#L154

**Comment by wtgodbe** [comment]
I got a successful `-allconfigurations` build locally using the changes from https://github.com/dotnet/arcade/pull/880

**Comment by wtgodbe** [comment]
Should be able to merge this once Arcade packages get published for https://github.com/dotnet/arcade/pull/880

**Comment by wtgodbe** [comment]
Waiting on https://github.com/dotnet/core-eng/issues/4259 and https://github.com/dotnet/core-eng/issues/4258 so we can get an official build out of Arcade

**Comment by wtgodbe** [comment]
@weshaggard I believe this should get a green CI now, any other concerns?

**Comment by wtgodbe** [comment]
> Deleting project workspace...Cannot delete workspace :java.nio.channels.ClosedChannelException

Not sure what's going on with the CI failures - @mmitche is this a known issue?

**Comment by wtgodbe** [comment]
@dotnet-bot test Windows x86 Release Build
@dotnet-bot test Packaging All Configurations x64 Debug Build
@dotnet-bot test NETFX x86 Release Build

**Comment by wtgodbe** [comment]
Looks like the failures are related to stuff in the new `eng` directory. Will rebase locally and figure it out

**Comment by wtgodbe** [comment]
> bin\testPkg\tools\dependencies.props(76,3): error MSB4019: The imported project "D:\j\workspace\windows-TGrou---0d2c9ac4\bin\testPkg\tools\eng\Versions.props" was not found.

Looks like we intentionally copy `dependencies.props` to the `bin` location, so it makes sense we wouldn't be able to find `eng\versions.props` when it's being searched for relative to `MsbuildThisFileDirectory`. Probably the best solution is to add a property `EngDir` in `Directory.build.props`, and look for the file relative to that.

**Comment by wtgodbe** [comment]
@weshaggard PTAL at https://github.com/dotnet/corefx/pull/32437/commits/9ec5155da1806e1e6ede80dc6aa3701fcbb3794a

**Comment by wtgodbe** [comment]
https://github.com/dotnet/corefx/pull/32437/files#diff-4b2f402501d23abbede6eac0f47783e4R60 didn't work, looks like `Directory.Build.props` isn't imported for package testing.

**Comment by wtgodbe** [comment]
I'll try copying `versions.props` to the package testing bin folder instead

**Comment by wtgodbe** [comment]
CI is green! Any other concerns before I merge @weshaggard?

File: src/shims/shims.proj [hunk 4]
```diff
@@ -36,30 +36,36 @@
...
-          Command="$(GenFacadesCmd) -contracts:&quot;@(NetFxContracts)&quot; @&quot;$(GenFacadesResponseFile)&quot;"
-          WorkingDirectory="$(ToolRuntimePath)" />
-
-    <Exec Condition="'@(NETStandardContracts)' != ''"
-          Command="$(GenFacadesCmd) -contracts:&quot;@(NETStandardContracts)&quot; @&quot;$(GenFacadesResponseFile)&quot;"
-          WorkingDirectory="$(ToolRuntimePath)" />
+    <GenFacadesTask
+      Condition="'@(NetFxContracts)' != ''"
+      Contracts="@(NetFxContracts)"
+      Seeds="@(GenFacadesSeeds, ',')"
```
Comment (weshaggard) [inline src/shims/shims.proj]: I don't think the `',')` is necessary, similar below. Feel free to clean up after merging.

File: src/shims/shims.proj [hunk 5]
```diff
@@ -2,6 +2,7 @@
 <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
   <Import Project="Directory.Build.props" />
 
+  <UsingTask TaskName="GenFacadesTask" AssemblyFile="$(PackagesDir)/$(MicrosoftDotNetGenFacadesPackage.ToLower())/$(MicrosoftDotNetGenFacadesPackageVersion)/tools/netcoreapp2.1/Microsoft.DotNet.GenFacades.dll" />
```
Comment (weshaggard) [inline src/shims/shims.proj]: Is this necessary? now that you are importing the targets in Directory.Build.targets?

**Comment by joperezr** [comment]
I didn't realize that we were already trying to move stuff to Arcade packages instead. I should probably do this in the future for all of the packaging tooling since that is all in arcade now.

**Comment by weshaggard** [comment]
>I didn't realize that we were already trying to move stuff to Arcade packages instead. I should probably do this in the future for all of the packaging tooling since that is all in arcade now.

FWIW your package changes were undone so when we get back to packaging work we need to undo that undo. See https://github.com/dotnet/arcade/pull/585

**Comment by joperezr** [comment]
Oh, thanks for letting me know. FWIW I agree with you that it's weird we decided to pull it out just because there were extra things to do to finish the end-to-end, but anyway...

File: src/shims/shims.proj [hunk 6]
```diff
@@ -73,6 +79,7 @@
         StrongNameSig="Open"
         MarkerFile="$(GenFacadesOutputPath)%(NETStandardContracts.Filename)%(NETStandardContracts.Extension).requires_signing" />
 
+    <MakeDir Directories="$(IntermediateOutputPath)" />
```
Comment (ViktorHofer) [inline src/shims/shims.proj]: @wtgodbe to be more correct this line should be:

```
<MakeDir Directories="$([System.IO.Path]::GetDirectoryName($(GenFacadesSemaphoreFile)))" />
```

**Pull request merged**